apis/apps/v1alpha1/componentdefinition_types.go

```diff
@@ -569,7 +580,82 @@ type ComponentLifecycleActions struct {
 	AccountProvision *LifecycleActionHandler `json:"accountProvision,omitempty"`
 }
 
-type ComponentSwitchoverSpec struct {
+type PostProvision struct {
+	LifecycleActionHandler `json:",inline"`
+
+	// ExecutePolicy defines the policy for executing the action.
+	// Cannot be updated.
+	// +optional
+	ExecutePolicy *PostProvisionExecutePolicy `json:"executePolicy,omitempty"`
+}
+
+type PreTerminate struct {
+	LifecycleActionHandler `json:",inline"`
+
+	// ExecutePolicy defines the policy for executing the action.
+	// Cannot be updated.
+	// +optional
+	ExecutePolicy *PreTerminateExecutePolicy `json:"executePolicy,omitempty"`
+}
+
+// ExecuteFrequencyPolicy defines the policy for executing frequency of the action.
+type ExecuteFrequencyPolicy string
+
+const (
+	OnceExecuteFrequencyPolicy   ExecuteFrequencyPolicy = "Once"
+	AlwaysExecuteFrequencyPolicy ExecuteFrequencyPolicy = "Always"
+)
+
+// PostProvisionExecuteAtType defines the type of the PostProvision action execution.
+type PostProvisionExecuteAtType string
+
+const (
+	PostProvisionImmediatelyExecuteType    PostProvisionExecuteAtType = "Immediately"
+	PostProvisionRuntimeReadyExecuteType   PostProvisionExecuteAtType = "RuntimeReady"
+	PostProvisionComponentReadyExecuteType PostProvisionExecuteAtType = "ComponentReady"
+)
+
+type PostProvisionExecutePolicy struct {
+	// ExecuteAt defines the time when the action will be executed.
+	// - Immediately: The PostProvision Action is executed immediately after the Component object is created,
+	// without guaranteeing the availability of the Component and its underlying resources. only after the Action is successfully executed will the Component's state turn to ready.
+	// - RuntimeReady: The PostProvision Action is executed after the Component object is created and once all underlying Runtimes are ready.
+	// only after the PostProvision is successfully executed will the Component's state turn to ready.
+	// - ComponentReady: The PostProvision Action is executed after the Component object is created and once the Component is ready.
+	// the execution process does not impact the state of the Component and the Cluster.
+	// Cannot be updated.
+	// +kubebuilder:default=ComponentReady
+	// +optional
+	ExecuteAt *PostProvisionExecuteAtType `json:"executeAt"`
+
+	// ExecuteFrequency defines the frequency of the action execution.
+	// - Once: The Action will be executed only once when the conditions of ExecuteAt are met.
+	// - Always: The Action will always be executed when the conditions of ExecuteAt are met, but only one Action task will be executed at a given time,
+	// Idempotency and correctness are ensured by the implementation of the custom action.
+	// Cannot be updated.
+	// +kubebuilder:default=Once
+	// +optional
+	ExecuteFrequency *ExecuteFrequencyPolicy `json:"executeFrequency"`
```

Comment:
> provision and termination itself indicate that it is a one-shot operation, unlike start and stop which may occur multiple times

I agree that preTerminate is a one-time operation, so I haven't defined an ExecuteFrequency for it. As for provision, I used that term because I couldn't think of a better word to convey the meaning of the action itself. Do you have any better suggestions? (Perhaps PostSetUp, PostPrePare, PostConfigure, or removing this definition and explicitly stating in the implementation that it's a one-time operation since there's currently no specific scenario that requires multiple executions.)